src/components/NoteProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, createContext, useContext, ReactNode } from 'react';
+import { usePathwayContext } from './PathwayProvider';
+import { usePatient } from './PatientProvider';
+import { getHumanName } from 'utils/fhirUtils';
+export interface Note {
+  patient: string;
+  date: string;
+  physician: string;
+  birthdate?: string;
+  mcodeElements: {
+    [key: string]: string;
+  };
+  pathway?: string;
+  node?: string;
+  status: string;
+  treatment?: string;
+  notes?: string;
+}
+
+interface NoteProviderProps {
+  children: ReactNode;
+  date: Date;
+  physician: string;
+}
+
+export const NoteContext = createContext<Note | null>(null);
+
+export const NoteProvider: FC<NoteProviderProps> = ({ children, date, physician }) => {
```

Comment:
This NoteProvider feels a little weird to me -- it's not actually providing the Note per se, it's providing a couple pieces of data to the note like the date (which should also probably be fetched at the time of the click instead of at render, though in practice those probably won't be all that different) and the clinician/user TO the note generation process. Based on an offline conversation with @KeeyanGhoreshi I don't think it's worth totally breaking up at this point since it works, but could we at least rename it like to `NoteDataProvider` or something like that?